app/services/message_channel_service.rb

```diff
@@ -20,27 +20,25 @@ class UnknownChannelError < StandardError
                          discussion_description_edited
                          motion_description_edited)
 
-  def self.subscribe_to(user:, channel: )
-    raise AccessDeniedError.new unless can_subscribe?(user: user, channel: channel)
-    PrivatePub.subscription(channel: channel, server: ENV['FAYE_URL'])
+  def self.subscribe_to(user:, model:)
+    raise AccessDeniedError.new unless can_subscribe?(user: user, model: model)
+    PrivatePub.subscription(channel: channel_for(model), server: ENV['FAYE_URL'])
   end
 
-  def self.can_subscribe?(user:, channel:)
-    key = channel_key(channel)
-    case channel_type(channel)
-    when 'group'         then user.ability.can? :see_private_content, Group.find(key)
```

Comment:
This was causing an N+1 when loading all the user's groups. I think it makes more sense for the rest of the world to not know about the concept of a 'channel' anyway; pass in the model and have the service handle it correctly.